cpu/rp2350/include/clock_conf.h

```diff
@@ -0,0 +1,113 @@
+#define XOSC_CTRL_ENABLE_LSB 12u
+#define XOSC_STATUS_STABLE_BITS 0x80000000u
+#define XOSC_HZ MHZ(12)
+#define REF_DIV 2
+#define VCO_FREQ 750000000u
+#define PD1 6
+#define PD2 2
+#define PLL_PWR_PD_BITS 0x00000001u
+#define PLL_PWR_VCOPD_BITS 0x00000020u
+#define PLL_CS_LOCK_BITS 0x80000000u
+#define PLL_PRIM_POSTDIV1_LSB 16u
+#define PLL_PRIM_POSTDIV2_LSB 12u
+#define PLL_PWR_POSTDIVPD_BITS 0x00000008u
+#define CLK_PERI_CTRL_ENABLE_BIT 1 << 11
+#define XOSC_FREQ 12000000
+#define POSTDIV1 6
+#define POSTDIV2 2
+#define CPUFREQ 125000000
+/** Maximum crystal frequency */
+#define CLOCK_XOSC_MAX MHZ(15)
+/** Minimum crystal frequency */
+#define CLOCK_XOSC_MIN MHZ(5)
+/** Crystal frequency */
+#define CLOCK_XOSC (XOSC_HZ)
+/** Minimum value of the post PLL clock divers */
+#define PLL_POSTDIV_MIN (1U)
+/** Maximum value of the post PLL clock divers */
+#define PLL_POSTDIV_MAX (7U)
+/** Minimum value of the PLL VCO feedback scaler */
+#define PLL_VCO_FEEDBACK_SCALE_MIN (16U)
+/** Maximum value of the PLL VCO feedback scaler */
+#define PLL_VCO_FEEDBACK_SCALE_MAX (320U)
+/** Minimum value of the clock divider applied before 
+ * feeding in the reference clock into the PLL */
+#define PLL_REF_DIV_MIN (1U)
+/** Minimum value of the clock divider applied before feeding in
+ *  the reference clock into the PLL */
+#define PLL_REF_DIV_MAX (1U)
+
+#if (PLL_VCO_FEEDBACK_SCALE_MIN < PLL_VCO_FEEDBACK_SCALE_MIN) || \
+    (PLL_VCO_FEEDBACK_SCALE_MAX > PLL_VCO_FEEDBACK_SCALE_MAX)
+#  error "Value for PLL_VCO_FEEDBACK_SCALE out of range, check config"
+#endif
+#if (PLL_REF_DIV_MIN < PLL_REF_DIV_MIN) || (PLL_REF_DIV_MAX > PLL_REF_DIV_MAX)
+#  error "Value for PLLxosc_sleep_REF_DIV out of range, check config"
+#endif
+#if (PLL_POSTDIV_MIN < PLL_POSTDIV_MIN) || (PLL_POSTDIV_MAX > PLL_POSTDIV_MAX)
+#  error "Value for PLL_POSTDIV out of range, check config"
+#endif
+#if ((CLOCK_XOSC > CLOCK_XOSC_MAX) || (CLOCK_XOSC < CLOCK_XOSC_MIN))
+#  error "Value for CLOCK_XOSC out of range, check config"
+#endif
+
+#define PDIV ((PD1 << PLL_PRIM_POSTDIV1_LSB) | (PD2 << PLL_PRIM_POSTDIV2_LSB))
+#define FBDIV ((VCO_FREQ / XOSC_HZ) / REF_DIV)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Configures the Crystal to run.
+ * @note The reference hardware manual suggests to use a 12 MHz crystal, which we use by default.
+ */
+void xosc_start(void);
+
+/**
+ * @brief   Stop the crystal.
+ */
+void xosc_stop(void);
+
+/**
+ * @brief   Sleep for a given number of cycles.
+ */
+void xosc_sleep(int32_t cycles);
+
+/**
+ * @brief   Reset the clock system.
+ *
+ * This function resets the clock system to a known state.
+ * It is recommended to call this function before configuring the clock system.
+ */
+void clock_reset(void);
+
+/**
+ * @brief Configures the XOSC and then sets CLK_SYS, PLL_SYS and CLK_PERI to it
+ * @warning Make sure to call clock_reset() before this function to reset the
+ * clock system
+ * @note RP2350 Docs Chapter 8, mostly 8.2 for more details
```

Comment:
```suggestion
 * @see RP2350 Docs Chapter 8, mostly 8.2 for more details
```
Did you intend to use `@see` here instead of `@note`?

Reply:
I have to be honest, I dont know which command does what :sweat_smile:

Reply:
![image](https://github.com/user-attachments/assets/5761522b-b709-4717-9b80-e4453316a6e0)

However, this code is kinda unaccessibly for Doxygen at the moment anyways, because there is no header.

Reply:
The code for that is
```
@note This is a note.

@see This is a see.
```

Doxygen automagically creates the "See also" text.